### PR TITLE
Bug 1122432 - [Stingray] UI-Transition - Banner

### PR DESCRIPTION
**Pull request by sean2449**

### CODE REVIEW

**Comment by try-server-hook** [comment]
[sean2449 Sung Chiu (sean2449)](https://github.com/sean2449) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=43cb0ba273fe)

File: tv_apps/smart-system/js/interactive_notifications.js [hunk 1]
```diff
@@ -273,21 +260,23 @@
...
-
-      if (!this.hasPendings() && window.AppWindowManager &&
-          AppWindowManager.getActiveApp()) {
-        // We need to stop KeyNavigator while we don't need it.
-        this._keyNavigator.stop();
-        // If there is active app, we need to focus it back.
-        AppWindowManager.getActiveApp().getTopMostWindow().focus();
-      }
+      this._banner.hide();
+      this._hideCallback = function () {
```
Comment (huchengtw-moz) [inline tv_apps/smart-system/js/interactive_notifications.js]: There is no need to create _hideCallback function dynamically. We should make it as a member function of InteractiveNotification.

The only thing we need to take care is `button` variable. It should still work while we move the following code out of _hideCallback:
```
if (msg.onClosed) {
  msg.onClosed(button);
}
```

File: tv_apps/tv_shared/components/smart-banner/script.js [hunk 2]
```diff
@@ -41,26 +48,51 @@ window.SmartBanner = (function(win) {
         // We only process 'background-color' because all states have this
         // change.
         if ((evt.propertyName !== 'background-color' &&
+             evt.propertyName !== 'transform' &&
+             evt.propertyName !== 'opacity' &&
              evt.propertyName !== 'width') ||
-            evt.target !== this) {
+             evt.target !== this) {
```
Comment (huchengtw-moz) [inline tv_apps/tv_shared/components/smart-banner/script.js]: No space needed becase we use `evt.target !== this` is aligned to (evt.propertyName......)

**Comment by try-server-hook** [comment]
[sean2449 Sung Chiu (sean2449)](https://github.com/sean2449) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=b6cb17001cc3)

File: tv_apps/smart-system/js/interactive_notifications.js [hunk 3]
```diff
@@ -101,6 +101,7 @@
     window.addEventListener('keyup', this);
 
     this._banner.addEventListener('transitionend', this);
```
Comment (huchengtw-moz) [inline tv_apps/smart-system/js/interactive_notifications.js]: remove this line.

File: tv_apps/tv_shared/components/smart-banner/style.css [hunk 4]
```diff
@@ -1,23 +1,29 @@
...
+ *   open: closed -> opening -> opened
+ *   close: opened -> closing -> closed
+ *   hide: current state -> hiding -> closed
+ *   flyOpen: flying -> closed -> opening -> opened
 **/
 
 smart-banner {
   outline: 0;
-  display: inline-block;
+  display: block;
```
Comment (huchengtw-moz) [inline tv_apps/tv_shared/components/smart-banner/style.css]: Why do we need block?

File: tv_apps/tv_shared/components/smart-banner/style.css [hunk 5]
```diff
@@ -1,23 +1,29 @@
...
   height: 7rem;
   /* left 2.8, icon 5.4, child panding 2 = 10.2rem */
```
Comment (huchengtw-moz) [inline tv_apps/tv_shared/components/smart-banner/style.css]: Please update the comment here to say why 11rem.

File: tv_apps/tv_shared/components/smart-banner/style.css [hunk 6]
```diff
@@ -102,7 +113,24 @@ smart-banner.opening:before {
...
+  opacity: 0;
+  transition-property: opacity;
+  transition-duration: 1s;
+}
+
+smart-banner.flying {
+  animation: flyAnimation 1s;
+}
+
+@keyframes flyAnimation {
```
Comment (huchengtw-moz) [inline tv_apps/tv_shared/components/smart-banner/style.css]: This kind of design gives us "shaking" behavior. This is unexpected. Does Scott accept this?

**Comment by try-server-hook** [comment]
[sean2449 Sung Chiu (sean2449)](https://github.com/sean2449) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=76b4c44e28ef)

File: tv_apps/tv_shared/components/smart-banner/script.js [hunk 7]
```diff
@@ -41,26 +46,49 @@ window.SmartBanner = (function(win) {
...
           this.fireEvent('closed');
         }
+
+        // go back to initial state after hiding
+        if (this.classList.contains('hiding')) {
+          this.classList.remove('opened');
+          this.classList.remove('opening');
+          this.classList.remove('closing');
+          this.classList.remove('flying');
+          this.init();
```
Comment (huchengtw-moz) [inline tv_apps/tv_shared/components/smart-banner/script.js]: Please move init to the place after remove('hiding');

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by try-server-hook** [comment]
[sean2449 Sung Chiu (sean2449)](https://github.com/sean2449) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=39a62e80752e)

**Pull request merged**